torch/_guards.py

```diff
@@ -40,19 +42,34 @@ def select(self, locals_, globals_):
         if self in (
             GuardSource.LOCAL,
             GuardSource.LOCAL_NN_MODULE,
+            GuardSource.LOCAL_FSDP_MODULE,
             GuardSource.SHAPE_ENV,
             GuardSource.RANDOM_VALUE,
         ):
             return locals_
-        if self in (GuardSource.GLOBAL, GuardSource.GLOBAL_NN_MODULE):
+        if self in (
+            GuardSource.GLOBAL,
+            GuardSource.GLOBAL_NN_MODULE,
+            GuardSource.GLOBAL_FSDP_MODULE,
+        ):
             return globals_
         raise NotImplementedError(str(self))
 
+    def is_fsdp_module(self) -> bool:
+        return self in (GuardSource.GLOBAL_FSDP_MODULE, GuardSource.LOCAL_FSDP_MODULE)
+
     def is_nn_module(self) -> bool:
-        return self in (GuardSource.GLOBAL_NN_MODULE, GuardSource.LOCAL_NN_MODULE)
+        return self.is_fsdp_module() or self in (
```

Comment:
nit: This is negligible overhead-wise, but you could consider swapping the order of the clauses in this `or` since `self.is_fsdp_module()` is much less likely to return `True` than the other clause.

Reply:
oh.. yea, i could swap it.  I actually did that first and i put it this way bc its easier to 'see' the fsdp part than if it's at the end.